### PR TITLE
SAL: Adding the ability to log user interactions with sites.

### PR DESCRIPTION
**Pull request by wojtekn**
#### Changes proposed in this Pull Request:

* This diff adds a plugin that logs different events like adding a post, updating a post, adding a comment, and updating a comment. It supports Simple and Jetpack sites.

It is a sync from D87117-code.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Comment by wojtekn** [comment]
The build was failing as I placed the `get_user_interactions()` method in the base site API class, and that function was using the WPCOM-specific logic that does not exist on the Jetpack side.

I removed it completely from the Jetpack scope and re-added it in a proper trait in WPCOM. (D87485-code)

File: projects/plugins/jetpack/sal/class.json-api-site-jetpack.php [hunk 1]
```diff
@@ -631,4 +631,12 @@ public function get_anchor_podcast() {
 		return $this->get_atomic_cloud_site_option( 'anchor_podcast' );
 	}
 
+	/**
+	 * Get user interactions with a site. Not used in Jetpack.
+	 *
+	 * @return null
+	 */
```
Comment (coder-karen) [inline projects/plugins/jetpack/sal/class.json-api-site-jetpack.php]:
```suggestion
	 * Get user interactions with a site. Not used in Jetpack.
	 *
	 * @see /wpcom/public.api/rest/sal/trait.json-api-site-wpcom.php.
	 *
	 * @return null
	 */
```

Minor suggestion, just to keep this comment in line with the others in terms of being able to trace the WPcom file in question better.

Comment (wojtekn) [reply projects/plugins/jetpack/sal/class.json-api-site-jetpack.php]: Thanks, good point, I've added the comment.

**Review by coder-karen — APPROVED** [review]
Works well 👍

**Comment by wojtekn** [comment]
r252399-wpcom

**Pull request merged**